src/pages/Admin/Charity/Permissions/Table.tsx

```diff
@@ -0,0 +1,129 @@
+import { useFormContext } from "react-hook-form";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useGetWallet } from "contexts/WalletContext";
+import TableSection, { Cells } from "components/TableSection";
+import { CheckField } from "components/form";
+import { getTypedKeys } from "helpers";
+import LockButton from "./LockButton";
+import { FormField, FormValues, UpdateableFormValues } from "./schema";
+
+const formValues: UpdateableFormValues = {
+  accountFees: {} as FormField,
+  beneficiaries_allowlist: {} as FormField,
+  contributors_allowlist: {} as FormField,
+  donationSplitParams: {} as FormField,
+  profile: {} as FormField,
+};
+const FORM_KEYS = getTypedKeys(formValues);
+
+export default function Table({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+
+  const {
+    register,
+    watch,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+
+  const isNormal = endow_type === "normal";
+
+  return (
+    <table
+      className={`${className} w-full text-sm rounded border border-separate border-spacing-0 border-prim`}
+    >
+      <TableSection
+        type="thead"
+        rowClass="bg-orange-l6 dark:bg-blue-d7 divide-x divide-prim"
+      >
+        <Cells
+          type="th"
+          cellClass="py-3 px-4 text-xs uppercase font-semibold text-left first:rounded-tl last:rounded-tr"
+        >
+          <>Action</>
+          <>Admin wallet</>
+          {isNormal ? <>Governance</> : null}
+          <>Delegate</>
+          <>Delegate address</>
+          <>Actions</>
+        </Cells>
+      </TableSection>
+      <TableSection
+        type="tbody"
+        rowClass="dark:bg-blue-d6 divide-x divide-prim"
+        selectedClass="bg-orange-l5 dark:bg-blue-d4"
+      >
+        {FORM_KEYS.map((fieldName) => {
+          const delegate = watch(`${fieldName}.delegate`);
+          const name = watch(`${fieldName}.name`);
+          const initDelegate = watch(`initialValues.${fieldName}.delegate`);
+          const initDelegateAddress = watch(
+            `initialValues.${fieldName}.delegate_address`
+          );
+          const initModifiable = watch(`initialValues.${fieldName}.modifiable`);
+          const modifiable = watch(`${fieldName}.modifiable`);
+
+          const isDisabled =
+            disabled ||
+            !initModifiable ||
+            (initDelegate &&
+              !!initDelegateAddress &&
+              initDelegateAddress !== wallet?.address);
```

Comment:
found a bug where on initial load,  `delegate` is checked but there's no initial delegate address 

<img width="1173" alt="image" src="https://user-images.githubusercontent.com/89639563/225825921-010c707b-dcde-42f8-97e0-f15c7082dc89.png">

REPRO: go to `admin/1/permissions`

Reply:
Unable to repro this, does the issue still occur?

Reply:
sorry wrong issue, marked this as resolved a while ago